### PR TITLE
Fix IP fragmentation checks for big-endian CPUs

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: src/net_builtin.c [hunk 1]
```diff
@@ -223,8 +223,8 @@ static struct ip *tx_ip(struct mg_tcpip_if *ifp, uint8_t *mac_dst,
   memcpy(eth->src, ifp->mac, sizeof(eth->src));  // Use our MAC
   eth->type = mg_htons(0x800);
   memset(ip, 0, sizeof(*ip));
-  ip->ver = 0x45;   // Version 4, header length 5 words
-  ip->frag = 0x40;  // Don't fragment
+  ip->ver = 0x45;             // Version 4, header length 5 words
+  ip->frag = mg_htons(0x40);  // Don't fragment
```
Comment (robertc2000) [inline src/net_builtin.c:227]: I think it should be mg_htons(0x4000) instead.

We want to set the first byte at the &ip->frag offset.

Thus, on a little endian mg_htons(0x4000) results in 0x0040, which means the first byte will be written with value 0x40.

On a big endian system instead, mg_htons(0x4000) would still be 0x4000, which means the first will be written with 0x40.

In both cases, we set the 'don't fragment' bit correctly.

**Review by robertc2000 — CHANGES_REQUESTED** [review]

File: src/net_builtin.c (hunk 1 above)
Comment (cpq) [reply src/net_builtin.c]: Makes sense, fixed

File: src/net_builtin.c [hunk 2]
```diff
@@ -784,7 +784,8 @@ static void rx_tcp(struct mg_tcpip_if *ifp, struct pkt *pkt) {
 }
 
 static void rx_ip(struct mg_tcpip_if *ifp, struct pkt *pkt) {
-  if (pkt->ip->frag & IP_MORE_FRAGS_MSK || pkt->ip->frag & IP_FRAG_OFFSET_MSK) {
+  uint16_t frag = mg_ntohs(pkt->ip->frag);
+  if (frag & IP_MORE_FRAGS_MSK || frag & IP_FRAG_OFFSET_MSK) {
```
Comment (robertc2000) [inline src/net_builtin.c:788]: The masks should be also be in big endian order.

```
#define IP_FRAG_OFFSET_MSK 0x1FFF
#define IP_MORE_FRAGS_MSK 0x2000
````

Comment (cpq) [reply src/net_builtin.c:788]: Done!

**Review by scaprile — APPROVED** [review]

**Pull request merged**